### ISSUE TITLE
Cannot resolve build dependencies

### ISSUE THREAD

**Issue by Dasug** [cynicaloptimist/improved-initiative#547]
When trying to set up a local instance of improved initiative (the current version from the ``development`` branch) I was getting an error message about a missing node.exe when using ``npm install``:
<details>
  <summary>npm install log</summary>

```
➜  improved-initiative git:(development) npm install                   
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'improved-initiative@2.18.0',
npm WARN EBADENGINE   required: { npm: '~6.13.4', node: '~12.14.1' },
npm WARN EBADENGINE   current: { node: 'v16.3.0', npm: '7.18.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'newrelic@6.2.0',
npm WARN EBADENGINE   required: { node: '>=8.0.0 <13.0.0', npm: '>=3.0.0' },
npm WARN EBADENGINE   current: { node: 'v16.3.0', npm: '7.18.1' }
npm WARN EBADENGINE }
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /home/dasug/dnd/improved-initiative/node_modules/node/bin/node.exe
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/home/dasug/dnd/improved-initiative/node_modules/node/bin/node.exe'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dasug/.npm/_logs/2021-06-17T21_02_53_922Z-debug.log
```
</details>

 As I'm on Linux and therefore obviously shouldn't have any node.exe files in the first place I tried regenerating the package-lock.json by using ``npm update``. However this also failed as there appears to be a version conflict in the dependencies:
<details>
  <summary>npm update log</summary>

```
➜  improved-initiative git:(development) npm update                    
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: improved-initiative@2.18.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.0.0-0" from enzyme-adapter-react-16@1.15.6
npm ERR! node_modules/enzyme-adapter-react-16
npm ERR!   dev enzyme-adapter-react-16@"^1.14.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/dasug/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dasug/.npm/_logs/2021-06-17T21_03_58_889Z-debug.log
```
</details>
<details>
  <summary>eresolve-report.txt</summary>

```
# npm resolution error report

2021-06-17T21:03:58.886Z

While resolving: improved-initiative@2.18.0
Found: react@17.0.2
node_modules/react
  react@"^17.0.2" from the root project

Could not resolve dependency:
peer react@"^16.0.0-0" from enzyme-adapter-react-16@1.15.6
node_modules/enzyme-adapter-react-16
  dev enzyme-adapter-react-16@"^1.14.0" from the root project

Fix the upstream dependency conflict, or retry
this command with --force, or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.

Raw JSON explanation object:

{
  "code": "ERESOLVE",
  "current": {
    "name": "react",
    "version": "17.0.2",
    "whileInstalling": {
      "name": "improved-initiative",
      "version": "2.18.0",
      "path": "/home/dasug/dnd/improved-initiative"
    },
    "location": "node_modules/react",
    "isWorkspace": false,
    "dependents": [
      {
        "type": "prod",
        "name": "react",
        "spec": "^17.0.2",
        "from": {
          "location": "/home/dasug/dnd/improved-initiative"
        }
      }
    ]
  },
  "currentEdge": {
    "type": "prod",
    "name": "react",
    "spec": "^17.0.2",
    "from": {
      "location": "/home/dasug/dnd/improved-initiative"
    }
  },
  "edge": {
    "type": "peer",
    "name": "react",
    "spec": "^16.0.0-0",
    "error": "INVALID",
    "from": {
      "name": "enzyme-adapter-react-16",
      "version": "1.15.6",
      "whileInstalling": {
        "name": "improved-initiative",
        "version": "2.18.0",
        "path": "/home/dasug/dnd/improved-initiative"
      },
      "location": "node_modules/enzyme-adapter-react-16",
      "isWorkspace": false,
      "dependents": [
        {
          "type": "dev",
          "name": "enzyme-adapter-react-16",
          "spec": "^1.14.0",
          "from": {
            "location": "/home/dasug/dnd/improved-initiative"
          }
        }
      ]
    }
  },
  "peerConflict": null,
  "strictPeerDeps": false,
  "force": false
}
```
</details>

It seems like you explicitly require ``react@"^17.0.2"``, however also require ``enzyme-adapter-react-16@"^1.14.0"``, which newest valid version itself requires ``react@"^16.0.0-0"``. As the two react version requirements are incompatible, npm cannot resolve the dependencies.

**Comment by cynicaloptimist** [comment]
Yeah, `package.json` in `development` is really screwed up right now. It's on my list of things to fix before the 3.0 release but I'd definitely welcome a PR that cleans things up a bit. I honestly am not sure what the right choices are for my `node` and `npm` dependencies.

**Comment by cynicaloptimist** [comment]
As far as Enzyme I'm actually working on porting over tests to react-testing-library so I can drop it altogether.
